### ISSUE TITLE
Need better Help dialog for unit variations

### ISSUE THREAD

**Issue by doofus-01** [wesnoth/wesnoth#6011]
### Game and System Information
Battle for Wesnoth v1.15.15+dev (f8a856bad7a-Modified) x86_64
1920x1015
### Describe the bug
Unit variations display in Help dialog look bad
 - Unit name (walking corpse, for example) is listed multiple times in tree view
 - Tree view line spacing jumps when a variation node is opened
 - There doesn't seem to be a convention on how to use `name` vs `variation_name`.  Comparing walking corpse to sand scuttler, probably either convention would work, but the GUI would have to deal with them differently.
 - Possibly the list of variations could be removed from the individual variation pages, since the base unit is linked to and the list is provided in the tree view.
 - Units with variations are bumped to the top of the list with a race in the tree view

### To Reproduce
Look at the unit descriptions in Help.

### Screenshots
![soulless-variation-help-pre](https://user-images.githubusercontent.com/13510196/128616062-f783803a-37e7-47d1-844d-2c165737fb0f.jpg)
![WC-variation-help-post](https://user-images.githubusercontent.com/13510196/128616064-12461102-2892-4564-a570-904d3a5c1573.jpg)

**Comment by stevecotton** [comment]
> Unit name (walking corpse, for example) is listed multiple times in tree view

FWIW, the code controlling that is here, but this patch also removes it from the title of the right-hand pane:
```
diff --git a/src/help/help_impl.cpp b/src/help/help_impl.cpp
index 8f13d32db7b..144002a53de 100644
--- a/src/help/help_impl.cpp
+++ b/src/help/help_impl.cpp
@@ -956,7 +956,7 @@ void generate_unit_sections(const config* /*help_cfg*/, section& sec, int /*leve
                for (const std::string &variation_id : type.variations()) {
                        // TODO: Do we apply encountered stuff to variations?
                        const unit_type &var_type = type.get_variation(variation_id);
-                       const std::string topic_name = var_type.type_name() + "\n" + var_type.variation_name();
+                       const std::string topic_name = var_type.variation_name();
                        const std::string var_ref = hidden_symbol(var_type.hide_help()) + variation_prefix + var_type.id() + "_" + variation_id;
 
                        topic var_topic(topic_name, var_ref, "");
```

**Comment by stevecotton** [comment]
Related issue:

* On the help page for abilities, the variation names are not shown. So when playing SotA, the page for the Plague ability is a list that says just "Walking Corpse" once for each variant.

**Comment by CelticMinstrel** [comment]
Would be nice if that could detect that every walking corpse variation has it and thus list just the base name, while still being able to list specific variations if there's an ability that only one variation has.

**Comment by doofus-01** [comment]
Yeah, if a variation has a new ability (not in base unit) but isn't listed, that's a bug.  But if the base unit and all the variations have a common ability, it would seem like clutter to list them all.  And yet, if the base unit has an ability that the variation lacks, it would be nice to have that shown somehow.

Maybe the variations could be listed under the base unit, at another indentation level in to show that they are a subcategory.

**Comment by CelticMinstrel** [comment]
Something like this:

* Walking Corpse (all)
* Giant Scorpion (Sand Scuttler, Jungle Striker)
* Horse (all except: Nightmare, Spirit Horse)

Probably use "all except" if more than 50% of the variations (including the base unit) have the ability. Or don't use it at all, if you don't like it.

**Comment by sevu** [comment]
> Unit name (walking corpse, for example) is listed multiple times in tree view

Having two lines in the left sidebar currently (type and variation name) causes all other units to take up more space as well, which looks bad. Just listing the variation in the sub-tree (the sub-tree below the type) would probably be an easy fix with impact.

**Comment by doofus-01** [comment]
Removing the `var_type.type_name() + "\n" + ` from line 1032 of help_impl.cpp to make it
`const std::string topic_name = var_type.variation_name();`
does fix the unit help tree
![Screenshot_20240120_help](https://github.com/wesnoth/wesnoth/assets/13510196/752411ef-b5c3-4984-b29f-91b78348579b)
but by itself it is probably not a solution for two reasons:
1.  There should probably be some check for it a variation has a distinct `name` rather than `variation_name`, assuming there is any good reason for that in UMC
2.  The simple deletion also loses the unit type name in the page title, which is pretty minor but might be noticed.

I have to leave this here, instead of a PR, because I don't know C++, so my simple fumbling at fixing these two issues fails.

**Comment by Wedge009** [comment]
At least some of the issues seem to be resolved with gui2 Help. @doofus-01 Do you see anything still outstanding?

**Comment by Wedge009** [comment]
With responses lacking, I had a look myself. The first two points seem to be resolved with GUI2.

In my Help, Sand Scuttler doesn't appear to have variations (maybe missing some unit encounters). But I don't see any issue with point 3.

The fourth point on removing variation links on variation pages seems fair to me - albeit not directly related to the GUI2 conversion so I think this issue can be removed from the GUI2 Help tracker.

I also don't see a problem with point 5 - if the unit variations are a collection I think it's reasonable to treat them in analogously to how directories and files are often displayed in a GUI-based file explorer application.

**Comment by doofus-01** [comment]
Sorry, I'd missed your question from March.  I'll check...

**Comment by doofus-01** [comment]
Using dab71068045, I agree the first two items are solved.  

I'm not sure about the third, maybe it's solved, but I don't see an apparent variation in core, aside from undead, ships, and dunefolk, and that involves taxonomy as well as real variations.  It does look a little odd to see all the "Dune this, Dune that, Dune, Dune, Dune" in the help treeview, but it's probably not worth the fix.

Items 4 and 5 seem same as ever.  I disagree about the file explorer analogy, since Soulless is a specific unit type with variations, not a directory or race.  The file explorer analogy probably does apply to the help taxonomy distinction, like the dunefolk.  Ideally, the variations node would have a different appearance than the taxonomy node, but I have no idea if if it's worth the effort.

**Comment by Wedge009** [comment]
So this can remain open for both points 4 and 5. I think it would have been better to log all of these separately, but if we just have two left then that's probably okay.
